packages/manager/modules/vps/src/dashboard/tile/configuration/upgrade/routing.js

```diff
@@ -0,0 +1,222 @@
+          // directly (without clicking on oui-radio in dashboard state)
+          if (upgradeModelPlan.planCode !== upgradeToPlan.planCode) {
+            set(configurationTile.model, upgradeType, upgradeToPlan);
+          }
+
+          return null;
+        });
+    },
+    onExit: (transition) => {
+      const configurationTilePromise = transition
+        .injector()
+        .getAsync('configurationTile');
+
+      return configurationTilePromise.then((configurationTile) => {
+        set(
+          configurationTile.model,
+          'memory',
+          VpsConfigurationTile.currentPlan,
+        );
+        set(
+          configurationTile.model,
+          'storage',
+          VpsConfigurationTile.currentPlan,
+        );
+      });
+    },
+    component: component.name,
+    resolve: {
+      upgradeType: /* @ngInject */ ($transition$) =>
+        $transition$.params().upgradeType,
+
+      upgradeStatus: /* @ngInject */ ($transition$) =>
+        $transition$.params().upgradeStatus,
+
+      upgradeOrderId: /* @ngInject */ ($transition$) =>
+        $transition$.params().upgradeOrderId,
+
+      redirectTo: () => 'vps.detail.dashboard',
+
+      upgradeInfo: /* @ngInject */ (
+        configurationTile,
+        serviceName,
+        upgradeStatus,
+        upgradeType,
+        vpsUpgrade,
+      ) => {
+        if (upgradeStatus === 'success') {
+          return true;
+        }
+
+        return vpsUpgrade.getUpgrade(
+          serviceName,
+          get(configurationTile.upgrades, `${upgradeType}.plan.planCode`),
+          { quantity: 1 },
+        );
+      },
+
+      hasDefaultPaymentMethod: /* @ngInject */ (ovhPaymentMethod) =>
+        ovhPaymentMethod.hasDefaultPaymentMethod(),
+
+      loaders: () => ({
+        upgrade: false,
+      }),
+
+      /* ----------  ouiModal layout resolves  ---------- */
+
+      type: /* @ngInject */ (upgradeStatus) => upgradeStatus,
+
+      heading: /* @ngInject */ (
+        $translate,
+        stateVps,
+        upgradeStatus,
+        upgradeType,
+        vps,
+      ) => {
+        if (upgradeStatus === 'success') {
+          return $translate.instant(
+            'vps_dashboard_tile_configuration_upgrade_success_title',
+          );
+        }
+
+        const translationKey = `vps_dashboard_tile_configuration_upgrade_${upgradeType}_action_title`;
+        const translationValues =
+          upgradeType === 'memory'
+            ? {
+                currentMemory: vps.ram.value,
+                nextMemory: vps.ram.value * 2,
+              }
+            : {
+                currentStorage: stateVps.model.disk,
+                nextStorage: stateVps.model.disk * 2,
+              };
+
+        return $translate.instant(translationKey, translationValues);
+      },
+
+      primaryLabel: /* @ngInject */ ($translate, upgradeStatus) => {
+        const translationKey =
+          upgradeStatus === 'success'
```

Comment:
Maybe `upgradeStatus === 'success' ` should be stored in a resolve variable, to be injected, as you use it more than once